common/src/main/java/com/microsoft/identity/common/internal/eststelemetry/EstsTelemetry.java

```diff
@@ -298,21 +287,25 @@ public void flush(final String correlationId, final String errorCode) {
         currentTelemetry.clearTelemetry();
         mTelemetryMap.remove(correlationId);
 
-
-        if (mLastRequestTelemetryCache != null) {
-            // remove old last request telemetry data from cache
-            mLastRequestTelemetryCache.clearAll();
-            // save new last request telemetry data to cache
-            mLastRequestTelemetryCache.saveRequestTelemetryToCache(lastTelemetry);
-        } else {
-            Logger.verbose(
+        if (mLastRequestTelemetryCache == null) {
+            Logger.warn(
                     TAG + methodName,
                     "Last Request Telemetry Cache object was null. " +
                             "Unable to save request telemetry to cache."
             );
+        } else if (eligibleToCache(lastTelemetry)) {
+            // remove old last request telemetry data from cache
+            mLastRequestTelemetryCache.clearAll();
+            // save new last request telemetry data to cache
+            mLastRequestTelemetryCache.saveRequestTelemetryToCache(lastTelemetry);
         }
     }
 
+    private boolean eligibleToCache(RequestTelemetry lastTelemetry) {
```

Comment:
nit: add some comment describing the reason we have to introduce this.

Reply:
Addressed